src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -36,42 +35,57 @@ jest.mock('utils/context', () => ({
 jest.mock('./components/PaymentInterfaces');
 
 describe('PaymentButton', () => {
+  let nbApiCalls: number;
   const formatPrice = (price: number, currency: string) =>
     new Intl.NumberFormat('en', {
       currency,
       style: 'currency',
     }).format(price);
 
   const Wrapper = ({
-    client = new QueryClient(),
-    order,
+    client = createTestQueryClient({ user: true }),
     children,
-  }: PropsWithChildren<{ order?: Order; client?: QueryClient }>) => (
-    <OrderProvider defaultOrder={order}>
+  }: PropsWithChildren<{ client?: QueryClient }>) => {
+    return (
       <IntlProvider locale="en">
-        <JoanieApiProvider>
-          <CourseProductProvider productId="" courseCode="00000">
-            <QueryClientProvider client={client}>{children}</QueryClientProvider>
-          </CourseProductProvider>
-        </JoanieApiProvider>
+        <QueryClientProvider client={client}>
+          <JoanieSessionProvider>
+            <CourseProductProvider productId="" courseCode="00000">
+              {children}
+            </CourseProductProvider>
+          </JoanieSessionProvider>
+        </QueryClientProvider>
       </IntlProvider>
-    </OrderProvider>
-  );
+    );
+  };
 
   beforeEach(() => {
     jest.useFakeTimers();
+    jest.clearAllTimers();
+    jest.resetAllMocks();
+
+    fetchMock.restore();
+    sessionStorage.clear();
+
+    // Joanie providers calls
+    fetchMock.get('https://joanie.test/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.test/api/v1.0/credit-cards/', []);
+    fetchMock.get('https://joanie.test/api/v1.0/addresses/', []);
+    nbApiCalls = 3;
   });
 
   afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
     cleanup();
-    fetchMock.restore();
-    jest.clearAllTimers();
-    jest.resetAllMocks();
   });
 
   it('should render a payment button', async () => {
     const product: Joanie.Product = ProductFactory().one();
-
+    fetchMock.get(
```

Comment:
Maybe I lack a bit of context: why do we need to rewrite this test? is it because of the workflow changes?

Reply:
Before: we simply create an `Order` with `OrderFactory`
Now: we mock the api call made by `useCurrentOrder`, it add the need of `JoanieSession` and all these small changes.